updater/lib/dependabot/job.rb

```diff
@@ -43,87 +45,132 @@ class Job
       dependency_groups
       dependency_group_to_refresh
       repo_private
-    ).freeze
-
-    attr_reader :allowed_updates,
-                :credentials,
-                :dependencies,
-                :existing_pull_requests,
-                :existing_group_pull_requests,
-                :id,
-                :ignore_conditions,
-                :package_manager,
-                :requirements_update_strategy,
-                :security_advisories,
-                :security_updates_only,
-                :source,
-                :token,
-                :vendor_dependencies,
-                :dependency_groups,
-                :dependency_group_to_refresh
+    ).freeze, T::Array[Symbol])
+
+    sig { returns(T::Array[T::Hash[String, T.untyped]]) }
+    attr_reader :allowed_updates
+
+    sig { returns(T::Array[T.untyped]) }
+    attr_reader :credentials
+
+    sig { returns(T.nilable(T::Array[String])) }
+    attr_reader :dependencies
+
+    sig { returns(T::Array[T.untyped]) }
+    attr_reader :existing_pull_requests
+
+    sig { returns(T::Array[T.untyped]) }
+    attr_reader :existing_group_pull_requests
+
+    sig { returns(T.any(Integer, String)) }
+    attr_reader :id
+
+    sig { returns(T::Array[T.untyped]) }
```

Comment:
```suggestion
    sig { returns(T::Array[Dependabot::Config::IgnoreCondition]) }
```

Reply:
Unfortunately that is not what this is, although I do wish it was.

This class takes the job input which is a big hash of everything, and sets the properties individually as variables and usually doesn't make them into a proper class. So they're still arrays of hashes of strings or nilable strings, etc etc. 😢